module/modmanager/manager.go

```diff
@@ -355,6 +366,152 @@ func (mgr *Manager) getModule(conf resource.Config) (*module, bool) {
 	return nil, false
 }
 
+var (
+	// oueTimeout is the length of time for which an OnUnexpectedExit function
+	// can execute blocking calls.
+	oueTimeout = 2 * time.Minute
+	// oueRestartInterval is the interval of time at which an OnUnexpectedExit
+	// function can attempt to restart the module process. Multiple restart
+	// attempts will use basic backoff.
+	oueRestartInterval = 5 * time.Second
```

Comment:
These durations are arbitrary; I am happy to change them according to anecdotal experience of module/hardware startup time. Keep in mind that we'll wait 5 seconds between the first and second attempt and 10 between the second and third.